cpp/src/gandiva/llvm_generator.h

```diff
@@ -49,10 +51,48 @@ class GANDIVA_EXPORT LLVMGenerator {
   static Status Make(std::shared_ptr<Configuration> config,
                      std::unique_ptr<LLVMGenerator>* llvm_generator);
 
+  static std::shared_ptr<Cache<BaseCacheKey, std::shared_ptr<llvm::MemoryBuffer>>>
+  GetCache();
+
   /// \brief Build the code for the expression trees for default mode. Each
   /// element in the vector represents an expression tree
   Status Build(const ExpressionVector& exprs, SelectionVector::Mode mode);
 
+  /// \brief Build the code for the expression trees for default mode with a LLVM
+  /// ObjectCache. Each element in the vector represents an expression tree
+  template <class KeyType>
+  Status Build(const ExpressionVector& exprs, SelectionVector::Mode mode,
+               GandivaObjectCache<KeyType>& obj_cache) {
+    selection_vector_mode_ = mode;
+
+    // Start measuring code gen time
+    auto begin = std::chrono::high_resolution_clock::now();
+    for (auto& expr : exprs) {
+      auto output = annotator_.AddOutputFieldDescriptor(expr->result());
+      ARROW_RETURN_NOT_OK(Add(expr, output));
+    }
+
+    // Stop measuring time, calculate the elapsed time and pass it to object cache
+    auto end = std::chrono::high_resolution_clock::now();
+    size_t elapsed =
```

Comment:
This is wrong. the elapse time was supposed to track the compilation time which is mainly done in FinalizeModule

Reply:
I know, but when the FinalizeModule is called, the LLVM will access the GandivaObjectCache file to execute the logic for the object code manipulation (to cache it or to get it from cache) so I need to set GandivaObjectCache before the FinalizeModule execution, and before the code sets the GandivaObjectCache it needs to know the elapsed time so the greedy dual cache logic works.

Do you have any suggestions how to proceed with that case?

Reply:
You can track the time inside the gandivaObjectCache; start timer in constructor and stop it in notifyObject method?
Also, its best not to knowingly proceed with incorrect solution if you are stuck somewhere.

Reply:
Sorry, it seemed right. But I fixed with your suggestion.